sklearn/datasets/_base.py

```diff
@@ -280,6 +285,67 @@ def load_data(module_path, data_file_name):
     return data, target, target_names
 
 
+def load_gzip_compressed_csv_data(data_module, data_file_name, **kwargs):
+    """Loads data_file_name from data_module with importlib.resources and np.loadtxt
+
+    1) Open resource file with `importlib.resources.open_binary`
+    2) Decompress file obj with `gzip.open`
+    3) Load decompressed data with `np.loadtxt`
+
+    Parameters
+    ----------
+    data_module : string or module
+        module where data lives;
+
+    data_file_name : string
+        Name of gzip-compressed csv (.csv.gz) file to be loaded from
+        data_module/data_file_name. For example 'diabetes_data.csv.gz'.
+
+    **kwargs : dict
+        Keyword arguments to be passed to `np.loadtxt`;
+        e.g. delimiter=","
+
+    Returns
+    -------
+    data : Numpy array
+        A 2D array with each row representing one sample and each column
+            representing the features and/or target of a given sample.
+    """
+    with importlib_resources.open_binary(
+        data_module, data_file_name
+    ) as compressed_file:
+        # GH20297: The following handling of the `encoding` kwarg
+        # should follow that in `np.loadtxt`
+        gzip_encoding = None
+        if "encoding" in kwargs and kwargs["encoding"] != "bytes":
+            gzip_encoding = kwargs["encoding"]
+
+        compressed_file = gzip.open(compressed_file, mode="rt", encoding=gzip_encoding)
+        data = np.loadtxt(compressed_file, **kwargs)
```

Comment:
I don't understand why we would pass the `encoding` kwarg to `np.loadtxt` if it has arleady been used to decode the text of `gzip.open`.

Either call `gzip.open` with `mode="rb"` and only pass encoding to `np.loadtxt` so that it handles the text decoding correctly itself or pass it only to `gzip.open` with `model="rt"` (in which case it would make sense to have `load_gzip_compressed_csv_data(data_module, data_file_name, encoding="utf-8", **kwargs)` to simplify the code to only forward `kwargs` to `np.loadtxt`.